src/commands/tasks/taskConfig.ts

```diff
@@ -0,0 +1,182 @@
+      throw new Error(`failed to parse file ${err.message}`);
+    }
+  } catch (err2) {
+    throw new Error(`Invalid config file: ${err2.message}`);
+  }
+
+  /* Resolve variables before doing any further processing */
+  taskConfig = replaceConfigVariables(taskConfig);
+
+  /* The config file path is relative to the location of the config file, so change it */
+  if (taskConfig.path) {
+    taskConfig.path = ospath.join(ospath.dirname(path), taskConfig.path);
+  }
+
+  /* For things which aren't mandatory, set sensible defaults anyway */
+  taskConfig.timeLimit ??= 60;
+  taskConfig.memoryLimit ??= 128;
+  taskConfig.environment ??= {};
+
+  /* Convert environment variables to the value format that the service expects */
+  if (taskConfig.environment) {
+    taskConfig.environment =
+      Object.entries(taskConfig.environment).reduce((prev:any, curr:any) => ({ ...prev, [curr[0]]: { value: curr[1] } }), {});
+  }
+  return taskConfig;
+}
+
+export async function* getValidatedConfigIterator(
+  { path, name, code, entryPoint, runtime, description, timeLimit, memoryLimit, executionPermission, env }: any
+): AsyncGenerator<TaskConfig> {
+  let taskConfig: TaskConfig = {};
+
+  if (path) {
+    /* Check if the specified path points to a directory or a file */
+    const stat = await fs.stat(path);
+    if (stat.isDirectory()) {
+      /* If we have a directory, read the contents and try to sync each directory. We will not overload with command line parameters */
+      const dirContents = await fs.readdir(path);
+
+      for (const file of dirContents) {
+        let filePath = ospath.join(path, file);
+        if ((await fs.stat(filePath)).isDirectory()) {
+          /* if directory, append task-config.json */
+          filePath = ospath.join(filePath, 'task-config.json');
+          const cfg = await loadSingleConfigFile(filePath);
+          if (cfg) {
+            await validateConfig(cfg);
+            yield cfg;
+          }
+        }
+      }
+      return;
+    }
+    /* If it's a file, it should be a configuration file so try to load it */
+    taskConfig = await loadSingleConfigFile(path);
+    if (!taskConfig) {
+      throw new Error("Path is a file but doesn't point to a valid configuration file");
+    }
+  }
+
+  /* At this point,
+        1. either a file was specified and we have an opportunity to overload parameters here through command line.
+        2. Or no file was specified and everything needs to be supplied separately
+   */
+  if (name) { taskConfig.name = name; }
+  if (code) { taskConfig.path = code; }
+  if (entryPoint) { taskConfig.entryPoint = entryPoint; }
+  if (runtime) { taskConfig.runtime = runtime; }
+  if (description) { taskConfig.description = description; }
+  if (timeLimit) { taskConfig.timeLimit = timeLimit; }
+  if (memoryLimit) { taskConfig.memoryLimit = memoryLimit; }
+  if (executionPermission) { taskConfig.executionPermission = executionPermission; }
+
+  taskConfig.timeLimit ??= 60;
```

Comment:
These default values are used 2 times in the file. In case these would change, we must change them on 2 locations.
Better would be to extract the default values to an object/config and use that set the default values.

For reference, the other usage of the variables is on line 115

Reply:
I noticed, but didn't change change it yet because it will go away in the next PR from ticket https://qompium.atlassian.net/browse/EXHCLI-26